eng/native/init-os-and-arch.sh

```diff
@@ -29,7 +29,7 @@ if [ "$os" = "SunOS" ]; then
     CPUName=$(isainfo -n)
 else
     # For rest of the operating systems, use uname(1) to determine what the CPU is.
-    CPUName=$(uname -m)
+    CPUName=${CPUName:=$(uname -m)}
```

Comment:
@jashook @hoyosjs I am pretty sure this is not what wee wamt longer term.

So the situation is when building natively on osx-arm64, we are currently using the osx-x64 dotnet core.  This means we are running msbuild in a Rosetta process.  When the shell is spawned, it apparently is also spawned in x64 shell. Here uname -m is returning `x64` or `amd64` and we get the wrong arch.  We get the rosetta arch and not the machine arch.

This hack allows CPUName as an environment variable to override `$(uname -m)`

We probably want a better environment variable if we want to go that way.